core/system/src/ipc.c

```diff
@@ -188,6 +188,18 @@ static bool ipc_io_is_ok(int box_id, const uvisor_ipc_io_t * io) {
            vmpu_buffer_access_is_ok(box_id, io->msg, io->desc->len);
 }
 
+static bool ipc_io_token_is_ok(uvisor_ipc_t * ipc, const uvisor_ipc_io_t * io) {
+    const uint32_t token = io->desc->token;
+    UvisorSpinlock * lock = &ipc->tokens_lock;
+    bool status = false;
```

Comment:
Maybe add `__builtin_pops(token) == 1` as well?

Reply:
Yeah, an individual request should only use a single token. This function isn't used to check when waiting for multiple requests to complete.